torch/distributed/distributed_c10d.py

```diff
@@ -2219,14 +2219,23 @@ def all_gather(tensor_list, tensor, group=None, async_op=False):
         work.wait()
 
 
-def _all_gather_base(output_tensor, input_tensor, group=None, async_op=False):
+def all_gather_into_tensor(output_tensor, input_tensor, group=None, async_op=False):
     """
-    Single tensor all gather. Gathers a single tensor from all ranks, and puts them in a single output tensor.
+    All ranks gather tensors from all other ranks and put them into a single
+    output tensor.
 
     Args:
-        output_tensor (Tensor): Output tensor. It should contain
-            correctly-sized tensors to be used for output of the collective.
-        input_tensor (Tensor): Tensor to be broadcast from current process.
+        output_tensor (Tensor): Output tensor to accommodate tensor elements
+            from all ranks. It must be correctly sized and have one of the
+            following forms:
+            (i) a concatenation of all the input tensors along the primary
+            dimension, see ``torch.cat()``;
+            (ii) a stack of all the input tensors along the primary dimension,
+            see ``torch.stack()``.
+            Examples below may better explain the supported output forms.
```

Comment:
the examples don't actually show a torch.cat or torch.stack? Should we either remove this, or add those in the examples doc?

Reply:
`torch.cat()` and `torch.stack()` are mentioned here to indicate how the output tensor may look like. They are more in the sense of "**as if** a torch.cat() is called on the input tensors." It doesn't mean the user needs to call them, because the `all_gather_into_tensor` would directly generate such form.

Reply:
makes sense, thanks!